lapce-data/src/editor/commands/indent_line.rs

```diff
@@ -19,62 +16,29 @@ pub struct IndentLineCommand<'a> {
 impl<'a> IndentLineCommand<'a> {
     pub fn execute<L: BufferDataListener>(
         self,
-        mut buffer: EditableBufferData<'a, L>,
+        buffer: EditableBufferData<'a, L>,
     ) -> Option<RopeDelta> {
         let Self {
             selection,
             cursor,
             tab_width,
         } = self;
 
-        let selection = selection
-            .unwrap_or_else(|| cursor.edit_selection(&buffer.buffer, tab_width));
-
-        let indent = buffer.indent_unit();
-        let mut edits = Vec::new();
-
-        let mut lines = HashSet::new();
-        for region in selection.regions() {
-            let start_line = buffer.line_of_offset(region.min());
-            let mut end_line = buffer.line_of_offset(region.max());
-            if end_line > start_line {
-                let end_line_start = buffer.offset_of_line(end_line);
-                if end_line_start == region.max() {
-                    end_line -= 1;
-                }
-            }
-            for line in start_line..end_line + 1 {
-                if lines.contains(&line) {
-                    continue;
-                }
-                lines.insert(line);
-                let line_content = buffer.buffer.line_content(line);
-                if line_content == "\n" || line_content == "\r\n" {
-                    continue;
-                }
-                let nonblank = buffer.first_non_blank_character_on_line(line);
-                let new_indent = if indent.starts_with('\t') {
-                    indent.to_string()
-                } else {
-                    let (_, col) = buffer.offset_to_line_col(nonblank, tab_width);
-                    " ".repeat(indent.len() - col % indent.len())
-                };
-                edits.push((Selection::caret(nonblank), new_indent));
-            }
-        }
-
-        let edits = edits
-            .iter()
-            .map(|(selection, s)| (selection, s.as_str()))
-            .collect::<Vec<(&Selection, &str)>>();
-
-        let delta = buffer.edit_multiple(&edits, EditType::InsertChars);
-        cursor.apply_delta(&delta);
-
-        Some(delta)
+        Some(indentation::create_multi_edits(
```

Comment:
I like the strategy here, but this makes reading the code a lot more difficult. For one, the implementation is now syntactically identical to outdenting, which is just confusing. I think this could be remedied by renaming `edit_one_line` to `indent/outdent_one_line`